server/channels/store/sqlstore/thread_store.go

```diff
@@ -1137,3 +1137,25 @@ func (s *SqlThreadStore) updateThreadParticipantsForUserTx(trx *sqlxTxWrapper, p
 
 	return nil
 }
+
+// UpdateTeamIdForChannelThreads updates the team id for all threads in a channel.
+// Specifically used when a channel is moved to a different team.
+// If a user is not member of the new team, the threads will be deleted by the
```

Comment:
Which user are we referring to? Any of the thread participants or the original poster?